iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -612,40 +615,43 @@ private Table getTable(org.apache.hadoop.hive.ql.metadata.Table hmsTable) {
 
   @Override
   public boolean canSetColStatistics(org.apache.hadoop.hive.ql.metadata.Table hmsTable) {
-    Table table = IcebergTableUtil.getTable(conf, hmsTable.getTTable());
-    return table.currentSnapshot() != null && getStatsSource().equals(HiveMetaHook.ICEBERG);
+    return getStatsSource().equals(HiveMetaHook.ICEBERG);
   }
 
   @Override
   public boolean setColStatistics(org.apache.hadoop.hive.ql.metadata.Table hmsTable, List<ColumnStatistics> colStats) {
     Table tbl = IcebergTableUtil.getTable(conf, hmsTable.getTTable());
-    return writeColStats(colStats.get(0), tbl);
+    return writeColStats(colStats, tbl);
   }
 
-  private boolean writeColStats(ColumnStatistics tableColStats, Table tbl) {
+  private boolean writeColStats(List<ColumnStatistics> colStats, Table tbl) {
     try {
       if (!shouldRewriteColStats(tbl)) {
-        checkAndMergeColStats(tableColStats, tbl);
+        checkAndMergeColStats(colStats, tbl);
       }
       // Currently, we are only serializing table level stats.
```

Comment:
this comment seems to be outdated with these changes?

Reply:
removed